src/Executor/Promise/Adapter/SyncPromiseAdapter.php

```diff
@@ -72,8 +74,13 @@ public function createRejected(Throwable $reason): Promise
         return new Promise($promise->reject($reason), $this);
     }
 
-    public function all(array $promisesOrValues): Promise
+    public function all(iterable $promisesOrValues): Promise
     {
+        assert(
+            is_array($promisesOrValues),
+            'SyncPromiseAdapter::all() requires arg to of array type'
+        );
```

Comment:
I think `iterable` should work here without modification

Reply:
There's indexing into array below `$result[$index]`

Reply:
So? `$result` is a newly built `array`

Reply:
editted the comment. There's restriction of key type to int string so array is reasonable.